### ISSUE TITLE
Bicep --what-if changes always detected on existing site config

### ISSUE THREAD

**Issue by kovalsky-tmz** [Azure/arm-template-whatif#265]
**Bicep version**
0.6.1

I would like to implement in pipeline dynamic check of --what-if changes on bicep deployment. The problem is there are couple places that bicep is always detecting as change:

Result of what-if:
```
  ~ Microsoft.Web/sites/xyz [2021-03-01]
    + properties.siteConfig.localMySqlEnabled:   false
    + properties.siteConfig.netFrameworkVersion: "v4.6"
    

  ~ Microsoft.Web/sites/xyz [2021-03-01]
    + properties.metadata: [
        0:

          name:  "CURRENT_STACK"
          value: "dotnetcore"

      ]

```
I tried to put netFrameworkVersion and localMySqlEnabled to both Microsoft.Web/sites (siteConfig section) and Microsoft.Web/sites/config separately/at same time,  it always detects these configs as a change.

About CURRENT_STACK=dotnetcore, its necessary to put becase otherwise it's creating App Service with null Stack Settings. That setting is also always detected as a change.

Please for ideas and help.
Thanks.

**Comment by tholmesdev** [comment]
Hello, any update on this? Thanks!

Reproduction: https://github.com/Azure/azure-cli/issues/24163

**Comment by jfranki** [comment]
In our case we always have this modification:
```
  ~ Microsoft.Web/sites/***************** [2022-09-01]
    + properties.siteConfig.localMySqlEnabled:     false
    + properties.siteConfig.minTlsVersion:         "1.2"
    + properties.siteConfig.netFrameworkVersion:   "4.6"
    + properties.siteConfig.use32BitWorkerProcess: false
``` 
It does not matter that those properties are explicitly defined in the template with those values (in an attempt to find a workaround).

**Comment by dlardo** [comment]
I'm getting something very similar when using Pulumi. I suspect the same root cause.

Partial output from a `pulumi preview --diff`:
```
~ siteConfig: {
    + localMySqlEnabled  : false
    + netFrameworkVersion: "v4.6"
  }
```

**Comment by mr-davidc** [comment]
I am also seeing this issue. This is my particular output and explicitly defining all options doesn't resolve the problem.

```
  ~ Microsoft.Web/sites/********************* [2024-04-01]
    + properties.siteConfig.ftpsState:                              "Disabled"
    + properties.siteConfig.ipSecurityRestrictions: [
        0:

          action:    "Allow"
          ipAddress: "LogicApps.AustraliaEast"
          name:      "LogicAppsAustraliaEastTraffic"
          priority:  100
          tag:       "ServiceTag"

        1:

          action:    "Deny"
          ipAddress: "0.0.0.0/0"
          name:      "DenyAllOtherTraffic"
          priority:  200

      ]
    + properties.siteConfig.localMySqlEnabled:                      false
    + properties.siteConfig.minTlsVersion:                          "1.2"
    + properties.siteConfig.netFrameworkVersion:                    "v8.0"
    + properties.siteConfig.publicNetworkAccess:                    "Enabled"
    + properties.siteConfig.scmIpSecurityRestrictionsDefaultAction: "Deny"
    + properties.siteConfig.use32BitWorkerProcess:                  false
```

**Comment by stap123** [comment]
We are seeing the same thing for a linux function consumption app. 

`minTlsVersion` is set explicitly and is not actually changing value. The others are not specified and never have been.

```
  ~ Microsoft.Web/sites/abc [2024-04-01]
    + properties.siteConfig.localMySqlEnabled:   false
    + properties.siteConfig.minTlsVersion:       "1.2"
    + properties.siteConfig.netFrameworkVersion: "v4.6"
```

**Comment by simonlrostron** [comment]
Can confirm the behavior reported by others with an app provisioned with API version `2024-11-01`.

_Whether unspecified or specified_ in the Bicep template, it seems that `properties.siteConfig.localMySqlEnabled` and `properties.siteConfig.netFrameworkVersion` are always reported as needing to be changed.

_When specified_, the following properties are always indicated as needing to be changed:

- `properties.siteConfig.ftpsState`
- `properties.siteConfig.minTlsVersion`
- `properties.siteConfig.use32BitWorkerProcess`
- `properties.siteConfig.vnetRouteAllEnabled`
- `properties.vnetRouteAllEnabled` 

Querying the raw resource config indicates that the properties in question are already set with the proposed value, with the exception of `vnetRouteAllEnabled`.

For whatever reason, I could not get outbound internet traffic routing enabled through Bicep using `vnetRouteAllEnabled`. Applying the change to enable outbound internet traffic using the Portal and then diffing the raw JSON shows only those two properties have changed. I saw the that Portal uses API version `2022-03-01` and sets `properties.vnetRouteAllEnabled`; by doing the same in Bicep I was able to get this setting enabled.

Not sure if that suggests there is something up with some of the API versions.

**Comment by torreymicrosoft** [comment]
Thanks for your patience, and sorry for the long wait on this one. It was recently picked up as one of the older open issues for the ARM Deployments team to review.

This is the `Microsoft.Web/sites` server side siteConfig default noise that #90 has been tracking: `properties.siteConfig.localMySqlEnabled`, `properties.siteConfig.netFrameworkVersion`, `properties.siteConfig.minTlsVersion`, `properties.siteConfig.use32BitWorkerProcess`, `properties.siteConfig.ftpsState` and friends reported as additions on every run, and, as several of you found, setting them explicitly in the template does not clear the diff.

Thanks to @stap123 and @simonlrostron for the recent confirmations, which carried this right through to `2024-11-01` and showed the raw resource already holding the proposed values. That is genuinely useful evidence and it is the same evidence #90 needs, so rather than keep two parallel threads on the same property set I am consolidating onto the older one.

Duplicate of #90.

One piece of your original report is a slightly different shape and I do not want it lost in the merge: `properties.metadata` with `CURRENT_STACK`. That is a metadata collection rather than a siteConfig default. If that is still costing you a false Modify on a current api-version, open a focused issue with the what-if output and I will take it separately.

Feel free to reopen if consolidating loses something for you.
